packages/icon/src/vaadin-icon.d.ts

```diff
@@ -68,6 +68,16 @@ declare class Icon extends ThemableMixin(ElementMixin(ControllerMixin(SlotStyles
    */
   svg: IconSvgLiteral | null;
 
+  /**
+   * The SVG source to be loaded as the icon. It can be:
+   * - an URL to a file containing the icon
+   * - an URL in the format "/path/to/file.svg#objectID", where the "objectID" refers to an ID attribute contained
+   *   inside the SVG referenced by the path. Note that the file needs to follow the same-origin policy.
+   * - a string in the format "data:image/svg+xml,<svg>...</svg>". You may need to use the "encodeURIComponent"
+   *   function for the SVG content passed
```

Comment:
nit: would be nice to have some examples of these on the dev page. Not a blocker for this PR. Can be added later.

Reply:
I can add them later.